packages/gatsby-theme-docs/gatsby-config.js

```diff
@@ -110,6 +119,14 @@ module.exports = (themeOptions = {}) => {
           path: path.resolve(`./src/content`),
         },
       },
+      // Topics content pages (.mdx)
+      {
+        resolve: 'gatsby-source-filesystem',
+        options: {
+          name: 'topics',
+          path: path.resolve(`./src/topics`),
+        },
+      },
```

Comment:
TBH this is really something it's not clear to me. 
This part wasn't here in the first place but I started seeing build errors like the following one if I don't add this entry:
<img width="1230" alt="Screenshot 2023-01-23 at 11 47 30" src="https://user-images.githubusercontent.com/106664971/214021595-3ab15dc2-9059-4e7b-ae6b-2aa6a42a8bd3.png">

Reply:
**my first guess, but we can talk and try out if you want: **

That's a content error that became visible because MDX v2 seems to be stricter. 

Files inside `/topics` are not pages but intended to be imported into and re-used in multiple actual MDX files that are pages.   We decided on "topics" as a naming convention because in tech writing the concept of putting together documentation from prefabricated content  blocks is called "topic based writing".  

A fragment that is included must not have frontmatter, frontmatter has no relevant use case there beyond being a comment explaining what the file is and where it is to be used.    The error you see is MDX trying to parse the three minuses as MDX because no preprocessor was in place to do all the magic.  

I cannot tell why adding it to the filesystem source fixed it to be honest.

Reply:
It's a good guess but it doesn't seem to be the case. I've just tried removing the frontmatter header, and the parsing error is still there:
<img width="1046" alt="Screenshot 2023-01-23 at 14 10 02" src="https://user-images.githubusercontent.com/106664971/214048006-907fd3f9-6bcf-4f09-94a2-2caeb4ca95e3.png">

So, the reason must be something else. Definitely related to mdx2 strictness

Reply:
Not necesssarily MDX v2 itself. Could also be the webpack loader not understanding that `.mdx` imports have to be compiled as MDX, not as JS.    It's still a parsing error, just that my guess was wrong what kind of parsing error